scheduler/core/time.py

```diff
@@ -0,0 +1,21 @@
+from datetime import datetime, timedelta
+from math import ceil
+
+from lucupy.minimodel import NightIndex
+
+
+class TimeHandler:
+
+    def __init__(self, num_nights_to_schedule, start, time_slot_length):
+        self.start = start
+        self.num_nights_to_schedule = num_nights_to_schedule
+        self.time_slot_length = time_slot_length
+
+    def dt2night_index(self, dt: datetime) -> NightIndex:
+        days = dt.day - self.start.day
+        if days > self.num_nights_to_schedule:
+            raise ValueError('Date outside of the number of night to schedule!')
```

Comment:
This seems like an error message that could contain more information in case it comes up... something like:

```python
max_day = self.start.day + datetime(days=self.num_nights_to_schedule - 1)
raise ValueError(f'Date {dt.strftime("%Y-%m-%d")} outside of valid range for scheduling: {self.start.strftime("%Y-%m-%d")} - {max_day.strftime("%Y-%m-%d")}')
```

I'm not quite sure what the type of `start` is.